pbxproj/pbxextensions/ProjectFiles.py

```diff
@@ -145,14 +145,15 @@ class ProjectFiles:
     def __init__(self):
         raise EnvironmentError('This class cannot be instantiated directly, use XcodeProject instead')
 
-    def add_file(self, path, parent=None, tree=TreeType.SOURCE_ROOT, target_name=None, force=True,
+    def add_file(self, path, name=None, parent=None, tree=TreeType.SOURCE_ROOT, target_name=None, force=True,
                  file_options=FileOptions()):
```

Comment:
@kronenthaler I was wondering if `name` should be added to `FileOptions` instead?

Let me know, I'm happy to refactor it this way.

Reply:
I will need to take a deeper look. At first glance, it should not break backwards compatibility given that provides a default value (and its management is consistent)

The deeper question i have is why is this necessary? Xcode infers the name of the file from the path, and that is what it adds to the project.

Reply:
@kronenthaler I need to take a deeper look as well myself 😅, I developed this on a Linux machine. It's like developing with your eyes closed.

I sent the original picture to my colleague (in another country) so he test and I found the following:

![image](https://github.com/kronenthaler/mod-pbxproj/assets/645156/9f6fca0b-e17f-4e23-a2c8-2b2a3d4d5dba)

You can see the conversation here: https://github.com/openedx/openedx-app-ios/pull/465#issuecomment-2188205860

So after a couple of attempts I noticed that XCode won't detect the translation files unless they're in the PBXVariantGroup and the name of the would be set to the locale instead of the path. Therefore, I chose to override the name and make it different.

My question, does it make sense to add the name in `FileOptions`? to ensure much more robust backward compatibility or the function parameter makes more sense?